cmd/admin_user_list.go

```diff
@@ -33,19 +33,59 @@ func runListUsers(c *cli.Context) error {
 		return err
 	}
 
-	users, err := user_model.GetAllUsers(ctx)
+	allUsers, err := user_model.GetAllUsers(ctx)
 	if err != nil {
 		return err
 	}
 
+	var users []*user_model.User
+	if c.IsSet("admin") {
+		for _, u := range allUsers {
+			if u.IsAdmin {
+				users = append(users, u)
+			}
+		}
+	} else {
+		users = allUsers
+	}
+
+	if globalBool(c, "output-json") {
+		type userInfo struct {
+			ID       int64  `json:"id"`
+			Name     string `json:"name"`
+			Email    string `json:"email"`
+			IsActive bool   `json:"is_active"`
+			IsAdmin  bool   `json:"is_admin"`
+			TwoFA    bool   `json:"two_fa,omitempty"`
+		}
+
+		userInfos := make([]*userInfo, 0, len(users))
+		for _, user := range users {
+			userInfos = append(userInfos, &userInfo{
+				ID:       user.ID,
+				Name:     user.Name,
+				Email:    user.Email,
+				IsActive: user.IsActive,
+				IsAdmin:  user.IsAdmin,
+			})
+		}
+
+		if c.IsSet("admin") {
```

Comment:
Why we need to detect if the flag has been set?

Reply:
2FA will only be set if "admin" is not set. I keep the code behavior the same as before.

https://github.com/go-gitea/gitea/blob/e08f1a9cbd582c73918e401eeba36261627f44a7/cmd/admin_user_list.go#L43-L56